be/src/util/block_compression.cpp

```diff
@@ -442,6 +451,103 @@ class ZstdBlockCompression final : public BlockCompressionCodec {
     size_t max_compressed_len(size_t len) const override { return ZSTD_compressBound(len); }
 };
 
+class GzipBlockCompression final : public ZlibBlockCompression {
+public:
+    static const GzipBlockCompression* instance() {
+        static GzipBlockCompression s_instance;
+        return &s_instance;
+    }
+
+    ~GzipBlockCompression() override = default;
+
+    Status init_compress_stream(z_stream& zstrm) const override {
+        zstrm.zalloc = Z_NULL;
+        zstrm.zfree = Z_NULL;
+        zstrm.opaque = Z_NULL;
+        auto zres = deflateInit2(&zstrm, Z_DEFAULT_COMPRESSION, Z_DEFLATED, MAX_WBITS + 16, 8, Z_DEFAULT_STRATEGY);
+        if (zres != Z_OK) {
+            return Status::InvalidArgument(
+                    strings::Substitute("Fail to do ZLib stream compress, error=$0, res=$1", zError(zres), zres));
+        }
+        return Status::OK();
+    }
+
+    Status compress(const Slice& input, Slice* output) const override {
+        std::vector<Slice> orig_slices;
+        orig_slices.emplace_back(input);
+        auto zres = ZlibBlockCompression::compress(orig_slices, output);
+        if (!zres.ok()) {
+            return Status::InvalidArgument(
+                    strings::Substitute("Fail to do ZLib compress, error=$0", zres.get_error_msg()));
+        }
+        return Status::OK();
+    }
+
+    Status decompress(const Slice& input, Slice* output) const override {
+        z_stream z_strm = {nullptr};
+        z_strm.zalloc = Z_NULL;
+        z_strm.zfree = Z_NULL;
+        z_strm.opaque = Z_NULL;
+
+        int ret = inflateInit2(&z_strm, MAX_WBITS + 16);
+        if (ret < 0) {
+            std::stringstream ss;
+            ss << "Failed to init inflate. status code: " << ret;
+            return Status::InternalError(ss.str());
```

Comment:
Could you print both error msg and return code, just like https://github.com/StarRocks/starrocks/pull/2739/files?show-viewed-files=true&file-filters%5B%5D=#diff-6d4580b48dea040705eab7a993d69c6ebc36ae8caaeeab3a5e9967abc72da1bfR311